source/server/http/admin.cc

```diff
@@ -492,6 +494,15 @@ Http::Code AdminImpl::handlerLogging(absl::string_view url, Http::HeaderMap&,
   return rc;
 }
 
+Http::Code AdminImpl::handlerMemory(absl::string_view, Http::HeaderMap&, Buffer::Instance& response,
+                                    AdminStream&) {
+  envoy::admin::v2alpha::Memory memory;
+  memory.set_currently_allocated_bytes(Memory::Stats::totalCurrentlyAllocated());
+  memory.set_heap_size_bytes(Memory::Stats::totalCurrentlyReserved());
```

Comment:
Should we use the same field names as tcmalloc to minimize any semantic gap?

Reply:
Those values are already exposed in `/stats` as:
```
server.memory_allocated: 70379952
server.memory_heap_size: 74448896
```
so I'd rather re-use the existing names (either `memory_allocated` and `memory_heap_size`  or just `allocated` and `heap_size`) than introducing new ones.

Reply:
Why not make a fresh start here? There are no legacy deps and we can aim for alignment with the underlying data source?

Reply:
I like the idea of `allocated` and `heap_size` to match the existing values.

Reply:
@htuch:

1. Using different names to represent the same thing only leads to confusion, i.e. is this the same value as the one in `/stats` and if so why does it have different name?
2. I don't believe anybody cares that those values come from `tcmalloc` and not OS or another data source, so the alignment here doesn't seem to provide any value.
3. Are we forever committed to `tcmalloc`? Other allocators (e.g. `jemalloc`) use different names for those stats.

But we're bike-shedding about using `currently_allocated_bytes` or `allocated` (since both `tcmalloc` and `/stats` use `heap_size`), so I'm not going to push too much against it.

Reply:
Arguable by renaming the `tcmalloc` stats names we are already violating (1). I think (2) is actually not quite right; every time I stare at `tcmalloc` stats, I need to remind myself specifically what they are measuring, as there is a lot of nuance on each of the different stats it emits (e.g. is this memory allocated, is it allocated but in thread-local cache, is this memory released but not back to the OS, etc.). 

I would probably vote to make this a `oneof` and export based on allocator the specific detailed, allocator defined stats. This is the most useful to the end user, since we don't need to be overly abstract and hide differences.

We can merge this as `v2alpha` and worry about the bikeshed when we lock down the API. @ambuc can you put in a TODO to consider adding more `tcmalloc` stats and exploring the naming tradeoffs?

Reply:
Sounds good -- agreed that these names can be misleading and should be bikeshedded to some degree. I'll merge this as-is and add a TODO for later.